source/client/process_impl.cc

```diff
@@ -261,6 +268,8 @@ class NighthawkServerInstance : public Envoy::Server::Instance {
   Envoy::Grpc::Context& grpc_context_;
   Envoy::Router::Context& router_context_;
   StatsConfigImpl stats_config_;
+  // Lifetime managed by ProcessImpl
```

Comment:
Do we need to retain the raw pointer? The parent unique pointer will go out of scope at some point, making this a dangling reference.

If we can - let's aintain the same lifetime on both the nighthawk_server and the server_factory_context by limiting them both to the stack of the single function that uses them.

Reply:
This has now become a reference.